### DISCUSSION TITLE
Two column layout with pre-defined windows, each window with a pre-defined position and size

### DISCUSSION THREAD

**Discussion by jose** [asmvik/yabai#1122] [Q&A] [answered]
I've been trying [yabai](https://github.com/koekeishiya/yabai) for a couple days now and I'm impress with such nice piece of software. I've been searching for a tiling window management for MacOS for a while, one that actually works and it's straightforward to get up-and-running, and I think I've found the one.

---

Now, to the question / ask for help.  What kind of commands would give me the following workflow, automatically?

Create a new workspace for a given directory/project and sets the name of that space as the name of the directory/project.  In the new space, create a two column layout.

```
  +---------- Finder -----------++------------ Chrome ------------+
  +                             ++                                +
  +-----------------------------++                                +
  +-- iTerm --++----- Atom -----++                                +
  +           ++                ++                                +
  +           ++                ++                                +
  +           ++                ++                                +
  +-----------++----------------++--------------------------------+
```

- Left Column:
  1. Open `Finder` in the project's directory.
     Window should be place on top, its height of 30, and its width should equal to `iTerm`'s width + `Atom`'s width.
  2. Open a new `iTerm2` window in the project's directory.
     Window should be place after `Finder`'s window on the left side and ideally should be 90x60.
  3. Open a new `Atom` window in the project's directory.
     Window should be place after `Finder`'s window on the right side and ideally should be 120x60.
- Right Column:
  1. Open a new, e.g., `Chrome/Safari` window.
     Window's width should be `screen width - dock width - left column width` and window's height should be `screen height - menu bar`.

Note: ideally, in the new space, any new window should (always) be open on the right column.

Any idea how could one achieve this?  Thanks in advance.

**Comment by lowell-lewis** [comment, marked as answer]
The below rule would map your chrome window basically the way you want. If you choose to do it with rules it's always going to open Chrome with the same layout. You can get even more control over it if you use signals (events) and scripting but that's more complicated. The grid is overlaid on your screen size. You just need to add 3 more rules for the other windows. You could also add space if you wanted them to appear on a specific space.

`yabai -m rule --add app="^Chrome$" grid=7:4:2:0:4:7`
